### ISSUE TITLE
Envelope won't get linear on "rev.d" hardware

### ISSUE THREAD

**Issue by pouek** [unwndevices/Raiju#1]
Hi,
my Raiju works 100% ok except for the curve potentiometer, which doesn't change anything : the envelope curves are always exponential, never linear or even in between.

Do I need to reflash the 1.00 firmware ?

Is there something else I can do ?

Can you also upload rev d PCB schem and boardview ?
Thanks

**Comment by PS03ST** [comment]
I have the same problem. I got it to work once, but it broke again. My module is also rev D, and draws twice as much current as advertised.

**Comment by pouek** [comment]
Ok, so it' sprobably a hardware fault.
Did you do something to make it work or it just happened ?

**Comment by PS03ST** [comment]
Updating the firmware trough the Raiju tool software seems to fix it, but it just stops working randomly after that.

**Comment by pouek** [comment]
Thanks, I'll try that and come back with the results. Hopefully a new firmware version can fix it. Or possibly a small hardware hack can do the job.
Did you notice a lower consumption when it was working again ?

**Comment by PS03ST** [comment]
Consumption did not change. It also create a low noise in my whole system. I'm also open for an hardware hack if there is one.

**Comment by pouek** [comment]
Ok, just to be sure we're in the same situation : everything works for you except the enveloppe stays exponential, as if the curve pot was dead ?

**Comment by pouek** [comment]
It looks like a power problem ideed, maybe bad filtering (capacitors?) on the internal 3.3V psu (w/ LDO)..
I'll report if my LDO gets significantly lower than the 80°C I've seen, and other useful stuff I may encounter.

**Comment by pouek** [comment]
Yeah it generates ~200mVpp on +12 and -12 rails. Tried without, nothing.
I guess adding bigger caps between those rails and the whole circuit would help. 
I'll test them and evenutally replace, or simpler, add some more in parallel…

**Comment by PS03ST** [comment]
I've tried adding capacitance on the power rail too, I've replaced the LDO, but nothing changed. I also noticed that while in boot mode (when you update the firmware) the module doesn't consume as much power.

**Comment by PS03ST** [comment]
And yes, everything works for me except the curve pot that stops working randomly. It just outputs an exponential curve.

**Comment by pouek** [comment]
Ok, thanks for your infos.
Then, maybe that 3v3 LDO is undersized ?

**Comment by pouek** [comment]
Edit : Based on rev e. schematics :
The main chip : ESP32WROOM32
Vdd 3.3V (⁺/₋0.3V)
Consumes 0.5A 

The LDO : LM1117-3.3 
Out 3V3 max 0.8A
And in our case, direct out to Vdd, then via a 2k2 fuse (?) to 3.3VA !
 - > +3.3
 - - > 2k2 (?) > +3.3VA

~~I'll try to divide by ten this "fuse"~~

EDIT : I realised (when testing) that the schematics are for rev e. and must differ fom our rev d. !

→In our case there is nothing between LDO's output and the HC4051s', PCM5102's and MCP6004's 3.3VA power rail. So we don't have a separate power rail. 

Maybe lifting pins plugged to 3.3VA and adding small smd current-limiting (and not fuse) resistors... ?
I don't believe there is a single spot where we can add a 2k2 (or something) resistor to limit the whole 3.3VA rail…
(I'll disassemble more and try to trace the  3.3V power rails… )

We'll have hard time finding a good way without the schematics and boardview !

@unwndevices : Help us please !
:]

**Comment by PS03ST** [comment]
maybe a stupid idea but would two LDO in parrallel help in this case ? (on top of eachother on the board)

**Comment by pouek** [comment]
![raiju3 3fix-WIP](https://github.com/user-attachments/assets/eb6b66fe-8116-4d54-9ed6-4b0d2ea77db0)

Here is a 1st insight.
Disassembling the front panel wasn't useful, it's packed so tight on the other side, I couldn't see anything.

**Comment by pouek** [comment]
Recap :
- 3.3V (yellow) is digital power
- 3.3VA (red) is analog power
- small dots are vias, little metallic holes bringing trace from other side
- for all red powers we can see a via, for some yellow we can't
- we should isolate red from yellow with 2k2 (?) current limiting resistor
- we must not forget each capacitor for each power pin

**Comment by pouek** [comment]
We can : 
- cut traces between red vias and the 1st cap they encounter, let's call em' our friends
- add that resistor to the positive (left) side of the electrolytic capacitor (marked 10 / 25V/ RV) underneath the LDO generating 3.3V
- connect the other side of that resistor to each friend (that were isolated from its via), with insulated bodge wire.

**Comment by pouek** [comment]
What do you say ?

**Comment by pouek** [comment]
> maybe a stupid idea but would two LDO in parrallel help in this case ? (on top of eachother on the board)

Well I feel it wouldn't ease the solution above that much.
I'll think of it tomorrow !

**Comment by pouek** [comment]
I see 2 vias just after the electrolytic cap under the LDO.
With a bit of luck, @unwndevices has separated the 2 power lines from that point.
Then we could cut only 1 trace, scrap both sides of the cut, and add a smd resistor on the 2 newly created pads !

I'll test that theory by cutting a random track of those 2 and see if each line is complete and isolated from the other !

**Comment by pouek** [comment]
It makes a little frog face. But I hesitate on opening it's mouth…
![Screenshot_20240731-215914_Camera](https://github.com/user-attachments/assets/608dbf91-dc01-4fe3-a029-241246456bc4)
Sorry, cut along the white line to separate the vias from the pad.

I removed the electrolytic so we can see exactly what's underneath.
It seems that pair of vias are really like one big via, (to let pass more power thru) so I guess they must also be connected together on the other side.

I'll make a break, maybe until @unwndevices comes to enlighten us !

**Comment by pouek** [comment]
P.S. : I tried replacing the LDO's output capacitor (still the same) with a 22uF based on the typical applications found in it's datasheet. Didn't seem to change anything. :/

**Comment by pouek** [comment]
Hi, I made a few more steps !
(Not sure if I should largely edit my previous posts to ease the reading of this journey)
Before I cut traces I made sure my previous plan was all good. And I've seen 2 errors :
U8, one of the two 74HC4051 chips is not on the VA 3.3 rail in the rev e. schem, and the offset pot VR2 should be.

**Comment by pouek** [comment]
I first cut traces between U9 and U5 decoupling caps and the vias freding them, added a THT 2k2 resistor to feed them both. At this point I noticed that these two chips isolated in my new VA rail now had only 2.5V of power, and that both the regulator and the ESP32 both overheated.
I then proceeded to continue by adding U3's pins 1 and 8 to this rail.
The envelope still always stay exponential…

Next I may add another regulator just in case the original isn't big enoughx now that I separated the rails, it's not a big process.
